File: backend/src/main/java/com/kh/backend/model/vo/Board.java
package com.kh.backend.model.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Board {

	private int boardNo;
	private int userNo;
	private String title;
	private String content;
	private char boardCode; // C 일반, S 중고, A 입양, M 실종, I 정보, E 이벤트,  N 공지
	private String enrollDate; // 일단 String
	private String modifyDate; // 일단 String
	private int views;
	private char status; // "Y - 정상 게시 D - 작성자가 삭제 B - 관리자가 블락 / 블라인드"
	private char detailStatus; // "P - 진행중(판매중, 분양중, 찾는중) D - 완료상태	N - 무상태(일반)"
	private int likes;
}

File: service-frontend/src/type/board.ts
// 보드에다가 쓸 타입변수들
export interface Board{
    // vo 클래스를 그대로 적으면 된다.
    boardNo : number,
    userNo : number,
    title : string,
    content : string,
    boardCode : string,
    enrollDate : string,
    modifyDate : string,
    views : number,
    status : string,
    detailStatus : string,
    likes : number,

    nickName : string,

    boardImg : BoardImg[]

    comment : Comment[]
    // 전부다 필수 값이다.
}

// 보드 이미지 타입 지정
export interface BoardImg{
    boardNo : number,
    changeName : string,
    imgNo : number,
    originName : string
}

// 보드 댓글 타입 지정
export interface Comment{
    commentNo : number,
    boardNo : number,
    userNo : number,
    content : string,
    commentData : string,
    modifyDate : string,
    status : string,
    refNo : number
}


// 보드에 대한 기본값 설정
export const initialBoard:Board = {
    boardNo : 0,
    userNo : 0,
    title : '',
    content : '',
    boardCode : 'C',
    enrollDate : '',
    modifyDate : '',
    views : 0,
    status : '',
    detailStatus : '',
    likes : 0,

    nickName : '',

    boardImg : [],
    comment : []
};

// 내보내기
export const initialBoardList:Board[] = [];

export interface SearchKeyword {
    title: Board['title']
};
